.github/workflows/scripts/slack_gem_notifications.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require 'httparty'
+
+# TODO: Add support for comparing latest and previous ruby versions
+# TODO: If possible, specify how many gem objects we recieve from Ruby Gems API. We don't
+# need a gem's entire history.
+
+WATCHED_GEMS = ARGV
+
+if WATCHED_GEMS.empty?
+  warn 'WATCHED_GEMS cannot be empty'
+  exit
+end
+
+def watch_gem(gem_name)
+  gem_info = HTTParty.get("https://rubygems.org/api/v1/versions/#{gem_name}.json")
+  latest_versions = []
+  gem_info.each do |gem|
```

Comment:
There are other methods in Ruby that return a new array containing the block's return values as items. You may be able to refactor this to use `#map` or `#collect` and avoid the array definition on line 20.

Reply:
^-- This would be a fun exercise. NOTE that the value returned by `break` will be used by `map`, so you'd have to do something like `break gem if gem['platform'] == 'ruby' && !latest_versions.empty?` to pull this off.